internal/ingress/controller/nginx.go

```diff
@@ -1084,37 +1074,65 @@ ratio = {{ .OtelSamplerRatio }}
 parent_based = {{ .OtelSamplerParentBased }}
 `
 
+func datadogOpentracingCfg(cfg ngx_config.Configuration) (string, error) {
+	jsn := map[string]interface{}{
+		"service": cfg.DatadogServiceName,
+		"agent_host": cfg.DatadogCollectorHost,
+		"agent_port": cfg.DatadogCollectorPort,
+		"environment": cfg.DatadogEnvironment,
+		"operation_name_override": cfg.DatadogOperationNameOverride,
+	}
+
+	// Omit "sample_rate" if the configuration's sample rate is unset (nil).
+	// Omitting "sample_rate" from the plugin JSON indicates to the tracer that
+	// it should use dynamic rates instead of a configured rate.
+	if cfg.DatadogSampleRate != nil {
+		jsn["sample_rate"] = *cfg.DatadogSampleRate
+	}
+
+	jsnBytes, err := json.Marshal(jsn)
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsnBytes), nil
+}
+
+func opentracingCfgFromTemplate(cfg ngx_config.Configuration, tmplName string, tmplText string) (string, error) {
+	tmpl, err := template.New(tmplName).Parse(tmplText)
+	if err != nil {
+		return "", err
+	}
+
+	tmplBuf := bytes.NewBuffer(make([]byte, 0))
+	err = tmpl.Execute(tmplBuf, cfg)
+	if err != nil {
+		return "", err
+	}
+
+	return tmplBuf.String(), nil
+}
+
 func createOpentracingCfg(cfg ngx_config.Configuration) error {
-	var tmpl *template.Template
+	var fileContent string
```

Comment:
Its final purpose is to be the contents of a file, but at this stage it's `conf`, `config` or at a stretch, `configData`
The part that makes it `fileContent` is the `os.WriteFile` at the end, not what it is until that point.

Reply:
will do